src/components/map/layers/NodeLayer.tsx

```diff
@@ -63,15 +40,19 @@ export default class NodeLayer extends Component<MarkerLayerProps> {
         let html;
         switch (node.type) {
         case NodeType.STOP: {
-            html = this.getNodeStopMarkerHtml(isSelected);
+            html = this.getMarkerHtml(isSelected ? s.stopMarkerHighlight : s.stopMarker);
             break;
         }
         case NodeType.CROSSROAD: {
-            html = this.getNodeCrossroadMarkerHtml(isSelected);
+            html = this.getMarkerHtml(isSelected ? s.crossroadMarkerHighlight : s.crossroadMarker);
             break;
         }
         default: {
-            throw new Error(`NodeType not supported: ${node.type}!`);
+            this.props.notificationStore!.addNotification({
+                message: `Solmun (id: '${node.id}') tyyppi on virheellinen`,
+                type: notificationType.WARNING,
+            });
```

Comment:
This addNotification brings an error msg to the console:

index.js:2178 Warning: Cannot update during an existing state transition (such as within `render` or another component's constructor). Render methods should be a pure function of props and state; constructor side-effects are an anti-pattern, but can be moved to `componentWillMount`.